op-e2e/actions/upgrades/interop_fork_test.go

```diff
@@ -0,0 +1,144 @@
+package upgrades
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/helpers"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/interop/dsl"
+	"github.com/ethereum-optimism/optimism/op-e2e/actions/upgrades/utils"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/derive"
+	"github.com/ethereum-optimism/optimism/op-node/rollup/event"
+	"github.com/ethereum-optimism/optimism/op-service/predeploys"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInteropUpgrade(gt *testing.T) {
+	t := helpers.NewDefaultTesting(gt)
+
+	// Run Isthmus until the interop upgrade
+	is := dsl.SetupInterop(t, dsl.SetInteropOffsetForAllL2s(uint64(15)))
+	actors := is.CreateActors()
+	actors.PrepareAndVerifyInitialState(t)
+
+	////////////////////////////
+	// Pre-upgrade Block Production
+	////////////////////////////
+
+	// Start op-nodes
+	actors.ChainA.Sequencer.ActL2PipelineFull(t)
+	actors.ChainB.Sequencer.ActL2PipelineFull(t)
+
+	rollupConfigA := is.Out.L2s[actors.ChainA.ChainID.String()].RollupCfg
+	rollupConfigB := is.Out.L2s[actors.ChainB.ChainID.String()].RollupCfg
+
+	// Verify Interop is not active at genesis yet
+	l2Head := actors.ChainA.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, rollupConfigA.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain A")
+	require.False(t, rollupConfigA.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain A")
+
+	l2Head = actors.ChainB.Sequencer.L2Unsafe()
+	require.NotZero(t, l2Head.Hash)
+	require.True(t, rollupConfigB.IsIsthmus(l2Head.Time), "Isthmus should be active at genesis in chain B")
+	require.False(t, rollupConfigB.IsInterop(l2Head.Time), "Interop should not be active at genesis in chain B")
+
+	// Submit all and mine for both chains
+	actors.ChainA.Sequencer.ActL2EmptyBlock(t)
+	actors.ChainB.Sequencer.ActL2EmptyBlock(t)
+
+	currentBlockHeaderA := actors.ChainA.SequencerEngine.L2Chain().CurrentBlock()
+	currentBlockHeaderB := actors.ChainB.SequencerEngine.L2Chain().CurrentBlock()
+	require.False(t, rollupConfigA.IsInterop(currentBlockHeaderA.Time),
+		"Interop should not be active at the first L1 inclusion block in chain A")
```

Comment:
Bit confused by what you mean by "L1 inclusion block" since there's no batch submission yet and we are looking at L2 blocks here. Did you mean to assert something about the L1 origin perhaps?

Reply:
Responded [here](https://github.com/ethereum-optimism/optimism/pull/15517#discussion_r2071892367).

Reply:
Note that Holocene activated most of it's logic by L1 inclusion block timestmap, which made sense for Holocene because it changed derivation rules. Interop activates execution layer changes by L2 block timestamp.

This should just read something like
```suggestion
	require.False(t, rollupConfigA.IsInterop(currentBlockHeaderA.Time),
		"Interop should not be active at the first non-genesis L2 block of chain A")
```

Reply:
TODO in follow-up